MQTTClient-C/src/MQTTClient.c

```diff
@@ -530,6 +532,17 @@ int MQTTUnsubscribe(MQTTClient *c, const char *topicFilter)
    else
       rc = FAILURE;
 
+   // We have to find the right message handler and remove it.
+   for (size_t i = 0; i < MAX_MESSAGE_HANDLERS; ++i)
+   {
+      if (c->messageHandlers[i].topicFilter != NULL && !strcmp(c->messageHandlers[i].topicFilter, topicFilter))
+      {
+         c->messageHandlers[i].topicFilter = NULL;
+         c->messageHandlers[i].fp = NULL;
```

Comment:
This only unsets the reference, right?

Reply:
Yes, other fields will be rewritten on the next subscribe call